### PR TITLE
JShell: Running "Execute Java Shell" from Maven project fails

### PR DESCRIPTION
**Pull request by matthiasblaesing**
When trying to run "Execute Java Shell" for a non-modular maven project, this fails because the VM can't be started.

The problem is, that NetBeans tries to start the VM with a module setup, but the project does not provide one (it is non-modular).

Please see individual commits for detailed explanation. 

Closes: #6692

### CODE REVIEW

**Comment by matthiasblaesing** [comment]
@sdedic the core of this change is in [1a41b91](https://github.com/apache/netbeans/pull/6905/commits/1a41b91367d9a511b5d4f99a068098b11c0a17ef). You touched that code in 2017 in a commit titled  "#270831: quoted possible spaces in agent specification / other arguments generated by JShell", I don't understand why in that changeset `isModularProject` was modified.

Could you please check if this update make sense to you?

File: java/jshell.support/src/org/netbeans/modules/jshell/editor/CompletionFilter.java [hunk 1]
```diff
@@ -55,6 +61,29 @@
...
+
+    private static final MethodHandle MH_GET_TYPE;
+    private static final MethodHandle MH_GET_CHARACTERS;
+
+    static {
+        MethodHandle getTypeBuilder = null;
+        MethodHandle getCharactersBuilder = null;
+        Lookup lookup = MethodHandles.lookup();
+        try {
+            getTypeBuilder = lookup.findVirtual(DocTrees.class, "getType", MethodType.methodType(TypeMirror.class, DocTreePath.class));
```
Comment (sdedic) [inline java/jshell.support/src/org/netbeans/modules/jshell/editor/CompletionFilter.java:74]: I'd suggest a more prominent LOG message, as `getType()` method will NPE if this `findVirtual` fails.
I am not very familiar with shimming different javac versions - @jlahoda should definitely review this.

File: java/jshell.support/src/org/netbeans/modules/jshell/tool/JShellTool.java [hunk 2]
```diff
@@ -1691,50 +1754,60 @@ boolean cmdClasspath(String arg) {
...
-                        fluff("Event debugging on");
-                        break;
-                    default:
-                        hard("Unknown debugging option: %c", ch);
-                        fluff("Use: 0 r g f c d");
-                        return false;
+        try {
+            if (arg.isEmpty()) {
+                debug = !debug;
+                MH_SET_DEBUG_FLAGS.invoke(state, debug ? DBG_GEN : 0);
```
Comment (sdedic) [inline java/jshell.support/src/org/netbeans/modules/jshell/tool/JShellTool.java:1760]: maybe guard against missing method ?

File: java/jshell.support/src/org/netbeans/modules/jshell/support/ShellSession.java [hunk 3]
```diff
@@ -944,8 +945,34 @@ public void notifyClosed(JShellEnvironment env, boolean remote) {
...
+            processOutput = new StringBuilder();
+            // VMStartException is raised when target VM fails to start. At time
+            // of writing com.sun.tools.jdi.AbstractLauncher does not read the
+            // process output, so diagnostic output is lost, try to recover it.
+            assert !startException.process().isAlive() : "Process expected to be not alive";
+            ByteArrayOutputStream baos = new ByteArrayOutputStream();
+            try {
+                startException.process().getInputStream().transferTo(baos);
+            } catch (IOException ex) {}
+            processOutput.append("Standard output:\n");
```
Comment (sdedic) [inline java/jshell.support/src/org/netbeans/modules/jshell/support/ShellSession.java:960]: L10n as it goes to UI ?

**Comment by lahodaj** [comment]
(I'll (try to) take a look tomorrow, sorry.)

File: java/jshell.support/src/org/netbeans/modules/jshell/support/ShellSession.java (hunk 3 above)
Comment (matthiasblaesing) [reply java/jshell.support/src/org/netbeans/modules/jshell/support/ShellSession.java]: Valid point - updated via 25ee1a49b5838031bd0d52294d66628ca43512a8

File: java/jshell.support/src/org/netbeans/modules/jshell/tool/JShellTool.java (hunk 2 above)
Comment (matthiasblaesing) [reply java/jshell.support/src/org/netbeans/modules/jshell/tool/JShellTool.java:1767]: Thanks for bringing this up. I added a null guard, that gives info to the user. While checking/testing this, I noticed, that my testing was tainted by modifications to the `netbeans.conf`. The package `jdk.internal.jshell.debug` needs to be exported to the unnamed module, that necessary change was also added.

File: java/jshell.support/src/org/netbeans/modules/jshell/editor/CompletionFilter.java (hunk 1 above)
Comment (matthiasblaesing) [reply java/jshell.support/src/org/netbeans/modules/jshell/editor/CompletionFilter.java:74]: Raised log level to warning - in a normal situation `DocTrees` is loaded from nb-javac and thus the two methods are present and can be resolved here. If it fails the methods are not present and thus can't be called in the first place, so I think this should be safe.

**Review by lahodaj — APPROVED** [review]
Makes sense to me.

**Comment by matthiasblaesing** [comment]
@sdedic @lahodaj thank you both for review.

I addressed the review comments, squashed the updates into their base counterparts. github actions is still green, so I'll merge.

**Pull request merged**